consumer/consumererror/retryable.go

```diff
@@ -15,17 +15,24 @@
 package consumererror // import "go.opentelemetry.io/collector/consumer/consumererror"
 
 import (
+	"time"
+
 	"go.opentelemetry.io/collector/pdata/plog"
 	"go.opentelemetry.io/collector/pdata/pmetric"
 	"go.opentelemetry.io/collector/pdata/ptrace"
 )
 
 type retryable[V ptrace.Traces | pmetric.Metrics | plog.Logs] struct {
 	error
-	data V
+	delay time.Duration
+	data  V
```

Comment:
We could also potentially add a `successful` count to this to indicate that it is a retryable partial success-style error. While this wouldn't be useable for OTLP partial success messages, other formats that support a partial success concept while allowing retries of some or all of the data could use it.

Reply:
Following the discussion on adding retryable partial success responses to OTLP, I think we should leave this out until there's a concrete need.